### DISCUSSION TITLE
[Idea] Online view of ruby sources?

### DISCUSSION THREAD

**Discussion by rubyFeedback** [rubygems/rubygems.org#2762] [Ideas]
Hey folks,

This is mostly just a rough idea, sorry about that; feel free to close it.

Background: since some time, some weeks or perhaps already a few months,
we can see online the diff between two gems. This is an external site for
"Review changes", such as for rack:

https://my.diffend.io/gems/rack/2.2.2/2.2.3

This is nice because we can quickly see what changed, without having to open
the .gem file at hand. So mostly a convenience feature IMO.

I was thinking "what if we could also browse the source, as-is"? Like on github.

Now, I understand that most gems are on github, so they have that feature. I
understand that. But not everyone is using github, so for these gems it may be
useful to be able to view the source.

This does not necessarily have to be enabled at rubygems.org; an external
site could do this too such as how my.diffend.io already does this for different
versions (and perhaps we can already view the source through it completely,
even without a diff). I am more thinking of a new entry called "Browse Source"
or "View Source" or something. Or a view such as the one available on
github and gitlab, but for rubygems.org. Anyway, this is just a rough idea, 
please don't worry too much. I just thought it may be something that could
be useful to have, even though we already sort of have that via diffend
view, as-is.

**Comment by sonalkr132** [comment]
sorry, I don't think we would consider showing the content of gem related to the functioning of rubygems.org. if someone was to implement this as a third-party website, we would be happy to add its link.
